easybuild/easyconfigs/e/Extrae/Extrae-4.2.0-fix-hw-counters-checks.patch

```diff
@@ -0,0 +1,96 @@
+diff -Nru extrae-4.2.0.orig/tests/functional/hw-counters/check_Extrae_PAPI_TOT_CYC.sh extrae-4.2.0/tests/functional/hw-counters/check_Extrae_PAPI_TOT_CYC.sh
+--- extrae-4.2.0.orig/tests/functional/hw-counters/check_Extrae_PAPI_TOT_CYC.sh	2024-07-01 16:12:03.454963219 +0200
++++ extrae-4.2.0/tests/functional/hw-counters/check_Extrae_PAPI_TOT_CYC.sh	2024-07-02 10:26:41.603395579 +0200
+@@ -11,6 +11,20 @@
+ ../../../src/merger/mpi2prv -f TRACE.mpits -o ${TRACE}.prv
+ 
+ # Check
++if ! command -v papi_avail &> /dev/null
++then
++        echo "papi_avail could not be found"
++        exit 0
++fi
++
++PAPI_TOT_CYC_available=`papi_avail | grep PAPI_TOT_CYC | awk '{print $3}'`
```

Comment:
I think we need to see what the result of this command is on the failing systems

Reply:
Yes, absolutely. 
@bedroge, Can you check it on your laptop?

Reply:
See my message below (https://github.com/easybuilders/easybuild-easyconfigs/pull/20690#issuecomment-2203125685). Both commands give the same output on our cluster, where the tests do complete successfully.

Reply:
And the error in the log is this one, right?: 
`Error! Could not find 'PAPI_TOT_INS'`
This message is printed by `CheckEntryInPCF ${TRACE}.pcf PAPI_TOT_INS`, which means that it is not able to find any reference to PAPI_TOT_INS in the generated trace file, which is _REALLY_ weird if PAPI_TOT_INS is available on the system...

Looking into the log, my bet is that this message at the beginning: `Extrae: PAPI_start failed to start eventset 1 on thread 0! (error = -1)` points to the root of the problem. For some reason, Extrae is not able to use/initialize PAPI and this causes the counters not being traced.

Reply:
I have pushed a small patch that hacks Extrae's source code. It will not solve anything but the message `Extrae: Error! Hardware counter PAPI_TOT_INS (0x80000032) cannot be added in set 1 (task 0, thread 0)` will be more informative.
@bedroge , can you try it on your laptop to see what the message says?

Reply:
```
Extrae: Error! Hardware counter PAPI_TOT_INS (0x80000032) cannot be added in set 1 (task 0, thread 0) because of -15
```

Reply:
Well, at least this provides some information. The call inside the Extrae code that is causing the problem is this one:
https://github.com/bsc-performance-tools/extrae/blob/7b755dc715003428c877d766e1415eea179c3912/src/tracer/hwc/papi_hwc.c#L612
This call is returning a -15 instead of the expected 0. According to:
https://github.com/icl-utk-edu/papi/blob/3974391b56d511a5551ff04ee42c7dd6f9410851/src/papi.h#L270C1-L270C81
this -15 value corresponds to `#define PAPI_EPERM      -15    /**< Permission level does not permit operation */`

Reply:
This could be related to the following: 
https://stackoverflow.com/questions/51911368/what-restriction-is-perf-event-paranoid-1-actually-putting-on-x86-perf
What value do you have in your `/proc/sys/kernel/perf_event_paranoid` ?

Reply:
It's 4, and it looks like that means that kernel profiling is not allowed?

Reply:
Yes, actually: 
>= 1: Disallow CPU event access
Which is the one we are interested in, I guess. Can you try setting it to -1 and see what happens?

Reply:
Yes, that worked! :tada:

Reply:
Yes!!! , I saw it :)